server/contexts/ctxerr/ctxerr.go

```diff
@@ -0,0 +1,78 @@
+// Package ctxerr provides functions to wrap errors with annotations and
+// stack traces, and to handle those errors such that unique instances of
+// those errors will be stored for an amount of time so that it can be
+// used to troubleshoot issues.
+//
+// Typical uses of this package should be to call New or Wrap[f] as close as
+// possible from where the error is encountered (or where it needs to be
+// created for New), and then to call Handle with the error only once, after it
+// bubbled back to the top of the call stack (e.g. in the HTTP handler, or in
+// the CLI command, etc.). It is fine to wrap the error with more annotations
+// along the way, by calling Wrap[f].
+package ctxerr
+
+import (
+	"context"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server/errorstore"
+	"github.com/pkg/errors"
+	"github.com/rotisserie/eris"
+)
+
+type key int
+
+const errHandlerKey key = 0
+
+// NewContext returns a context derived from ctx that contains the provided
+// error handler.
+func NewContext(ctx context.Context, eh *errorstore.Handler) context.Context {
+	return context.WithValue(ctx, errHandlerKey, eh)
+}
+
+func fromContext(ctx context.Context) *errorstore.Handler {
+	v, _ := ctx.Value(errHandlerKey).(*errorstore.Handler)
+	return v
+}
+
+// New creates a new error with the provided error message.
+func New(ctx context.Context, errMsg string) error {
+	return ensureCommonMetadata(ctx, errors.New(errMsg))
+}
+
+// Wrap annotates err with the provided message.
+func Wrap(ctx context.Context, err error, msg string) error {
+	err = ensureCommonMetadata(ctx, err)
+	// do not wrap with eris.Wrap, as we want only the root error closest to the
```

Comment:
I don't follow the comment. Why are we using `pkg/errors.Wrap` and not `eris.Wrap` here (same on `Wrapf`). Does `pkg/errors.Wrap/f` do something better than `eris.Wrap/f` or has some behavior that we prefer?

Reply:
It has to do with how eris works internally and how we hash the errors to deduplicate them when stored - we should only capture the stack trace once, where the error actually happens, and that's the stack that will be hashed to identify the location of the error. When wrapping subsequently (when moving back up the stack), we should only annotate the error message, not capture more stack traces. In fact, your comment made me realize that we shouldn't even use pkg/errors here (as it also unnecessarily captures stack traces), just `fmt.Errorf` would be enough to properly wrap and annotate the error.